PhpAmqpLib/Wire/IO/StreamIO.php

```diff
@@ -154,6 +153,18 @@ public function connect()
         if ($this->keepalive) {
             $this->enable_keepalive();
         }
+
+        $options = stream_context_get_options($this->context);
+        if (isset($options['ssl']['crypto_method']) && $this->protocol === 'tcp') {
+            do {
+                $enabled = stream_socket_enable_crypto($this->sock, true);
+            } while ($enabled === 0);
```

Comment:
I agree, it looks pretty creepy. However, the documentation states that the possible return values are true, false and 0, and if it's 0, which it might be in the case of a non-blocking socket, we should retry the call.
Do you have an advice on how we should limit the loop here? And should we?

Reply:
Of course we should. Just add extra check for retry count and very small delay between retries.

Reply:
I used connection_timeout as enabling crypto is a part of the connection process. It should be more predictable and convenient.